marketplace/picture-optimizer.schema.json

```diff
@@ -0,0 +1,72 @@
+{
+    "$id": "https://schemas.aramis.group/marketplace/picture-optimizer.schema.json",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "operations": {
+            "type": "object",
+            "additionalProperties": false,
+            "description": "List of operations (filters) to apply on picture",
+            "properties": {
+                "cropping": {
+                    "type": "boolean"
+                },
+                "cropping-margin": {
+                    "type": "boolean"
+                },
+                "cropping-margin-ratio": {
+                    "type": "object",
+                    "description": "List of cropping margin auto",
+                    "additionalProperties": false,
+                    "properties": {
+                        "top" : {
+                            "type": "number"
+                        },
+                        "bottom" : {
+                            "type": "number"
+                        },
+                        "left" : {
+                            "type": "number"
+                        },
+                        "right" : {
+                            "type": "number"
+                        }
+                    },
+                    "minProperties": 4,
+                    "maxProperties": 4,
+                    "required": ["top", "bottom", "left", "right"]
+                },
+                "keywordsBlurring": {
+                    "type": "object",
+                    "description": "List of keywords for blurring",
+                    "additionalProperties": false,
+                    "properties": {
+                        "active": {
+                            "type": "boolean"
+                        },
+                        "keywords" : {
+                            "type": "object",
+                            "minProperties": 1
+                        }
+                    },
+                    "if": {"properties": {"active" : true}},
+                    "then": {"required": ["keywords"]},
+                    "minProperties": 2,
+                    "maxProperties": 2
+                },
+                "plate": {
+                    "type": "boolean"
+                },
+                "backgroundRemoval": {
+                    "type": "boolean"
+                }
+            },
+            "minProperties": 1,
```

Comment:
Is it possible to have no operation at all? Can we disable image processing altogether for a partner?

Reply:
yes